recipe/windows/test_nvcc.bat

```diff
@@ -71,3 +71,22 @@ if errorlevel 1 exit 1
 
 :: Try building something
 nvcc test.cu
+
+
+:: Try different CMake setups
+cd cmake-tests/
+
+for %%F in (
+    "-DWITH_FINDCUDA=ON -DWITH_FINDCUDATOOLKIT=OFF -DWITH_ENABLE_LANGUAGE=OFF"
+    "-DWITH_FINDCUDA=ON -DWITH_FINDCUDATOOLKIT=OFF -DWITH_ENABLE_LANGUAGE=ON"
+    "-DWITH_FINDCUDA=OFF -DWITH_FINDCUDATOOLKIT=ON -DWITH_ENABLE_LANGUAGE=ON"
```

Comment:
I think these are the 5 cases we want to test, and in each case we want to make sure that CUDA related paths are being set as expected, but that CMake is using our `nvcc` wrapper instead of the `nvcc` binary directly:
```suggestion
    "-DWITH_FINDCUDA=OFF -DWITH_FINDCUDATOOLKIT=OFF -DWITH_ENABLE_LANGUAGE=ON"
    "-DWITH_FINDCUDA=ON -DWITH_FINDCUDATOOLKIT=OFF -DWITH_ENABLE_LANGUAGE=OFF"
    "-DWITH_FINDCUDA=OFF -DWITH_FINDCUDATOOLKIT=ON -DWITH_ENABLE_LANGUAGE=OFF"
    "-DWITH_FINDCUDA=ON -DWITH_FINDCUDATOOLKIT=OFF -DWITH_ENABLE_LANGUAGE=ON"
    "-DWITH_FINDCUDA=OFF -DWITH_FINDCUDATOOLKIT=ON -DWITH_ENABLE_LANGUAGE=ON"
```

Reply:
Windows tests were not running because the nvcc wrapper returns right away and exits the script. Apparently this needs to be run with `call nvcc ...`. My love for cmd won't stop growing! 

I have amended the Windows tests now and added different generators too.